### PR TITLE
Update labextension build cli to include a parameter for setting the webpack publicPath option

### PR DESCRIPTION
**Pull request by thaddmt**
## Code changes

- Updated the labextension build cli options to support `public_webpack_path`
- Update `build-extension.ts` script to support the `public_webpack_path` option and sets an environment variable for the webpack cli to use
- Update `webpack.config.ext.ts` to read in `PUBLIC_WEBPACK_PATH` from env variable and to set it on the webpack config output options

### CODE REVIEW

**Comment by jasongrout** [comment]
The public webpack path is set dynamically at run time based on the information from page config. Why are you wanting to set it from the command line?

**Comment by jasongrout** [comment]
See https://github.com/jupyterlab/jupyterlab/pull/8861

**Comment by blink1073** [comment]
This is for dynamically loaded extensions whose static assets can be hosted elsewhere

**Comment by jasongrout** [comment]
Right now, it looks like this will compete with the existing public path setup and things will probably mess up. I think it would be better to work within that publicpath.js file that is generated in the webpack config.

I also think the option should not be named "webpack" - we haven't exposed that we are using webpack anywhere else.

File: builder/src/webpack.config.ext.ts [hunk 1]
```diff
@@ -179,7 +180,8 @@ module.exports = [
     },
     output: {
       filename: '[name].[chunkhash].js',
-      path: outputPath
+      path: outputPath,
+      publicPath: outputPublicPath
```
Comment (jasongrout) [inline builder/src/webpack.config.ext.ts:184]: Line 170 overrides this option. Please set this on line 170.

**Comment by thaddmt** [comment]
> Right now, it looks like this will compete with the existing public path setup and things will probably mess up. I think it would be better to work within that publicpath.js file that is generated in the webpack config.
> 
> I also think the option should not be named "webpack" - we haven't exposed that we are using webpack anywhere else.

That all makes sense. Will update the publicpath.js file and will rename the option

File: jupyterlab/labextensions.py [hunk 2]
```diff
@@ -190,9 +190,12 @@ def run_task(self):
 class BuildLabExtensionApp(BaseExtensionApp):
     description = "Build labextension"
 
+    static_path = Unicode('', config=True,
```
Comment (jasongrout) [inline jupyterlab/labextensions.py:193]: Do we want to default to `None`, since we are checking for `None` in `dynamic_labextensions.py`?

Comment (jasongrout) [reply jupyterlab/labextensions.py:193]: (I think it works now - I think it just passes the empty string all the way through to the check against the empty string - it's more a question of style)

**Comment by blink1073** [comment]
Congratulations on your first contribution to JupyterLab @thaddmt!  :tada:

**Pull request merged**